lib/most_frequent_integer.rb

```diff
@@ -3,5 +3,29 @@
 # exists, return nil.
 
 def find_most_frequent_integer(arr)
-  # put your code here
+  
+    #1. Create test array.
+    abs_arr = [1, 1, -1, -1, 2, 3, 4]
+    
+    #2. Create hash variable with default 0 value.
+    hashld = Hash.new 0
+    
+    #3. Loop through each array element, count them, and insert the final key/value combo into the hashld hash.
+        abs_arr.each do |ele|
+    	    hashld[ele] += 1
+        end
+    
+    #4. Hold max count in variable.
+    maxd = hashld.values.max
+    
+    #5. Create new hash containing keys with more than 1 count.
+    prearr = (hashld.select {|key,value| value == maxd})
+    
+    #6. Return array if size == 1, else return nil. 
+    if prearr.size == 1
+        prearr.keys
```

Comment:
`a_hash.keys` returns an array of key values. 

Even for a hash with only one key-value pair, like `a_hash = { "k" => "v" }`,
`a_hash.keys` returns `["k"]`

Because the tests are expecting the most-frequently-occurring integer itself as a return from this method (or `nil`), you'll have to do just a little more work here.